core/state.go

```diff
@@ -367,6 +368,34 @@ func (s *State) Class(classHash *felt.Felt) (*DeclaredClassDefinition, error) {
 	return class, err
 }
 
+func (s *State) CompiledClassHash(classHash *felt.Felt) (felt.Felt, error) {
+	classTrie, closer, err := s.classesTrie()
+	if err != nil {
+		return felt.Felt{}, err
+	}
+	defer closer()
+
+	casmHash, err := classTrie.Get(classHash)
+	if err != nil {
+		return felt.Felt{}, err
+	}
+
+	if casmHash.IsZero() {
+		return felt.Felt{}, errors.New("casm hash not found")
+	}
+
+	return casmHash, nil
+}
+
+func (s *State) CompiledClassHashV2(classHash *felt.Felt) (felt.Felt, error) {
```

Comment:
And this function.

The goal is to have any new code we write use the right types instead of the generic `felt.Felt`